### DISCUSSION TITLE
LilyGo T-Display S3 how to make code start on power-up ?

### DISCUSSION THREAD

**Discussion by gert4git** [espressif/arduino-esp32#7281] [General]
Hi,

Newbee for ESP23 here. Using this board:
https://github.com/Xinyuan-LilyGO/T-Display-S3
Board setting 'ESP32 S3 DEV'

I'm using Arduino IDE to write code. Once I have uploaded code and it works it is not starting again after power-cycle. I am familiar with arduino and adafruit feather in same IDE where uploaded code is persistent after power-cycle.

Is there a setting in IDE to make uploaded code reside into eeprom and launch after power-cycle? (for this board?)

And for sure that needs to be reversible to change code. I don't want this to be a oneway road. :-)

Thanks,
Gert

**Comment by Jason2866** [comment]
It is a standardl Esp32-S3 so your sketch is stored in flash memory which will not be lost when power is switched off. Normal Arduino sketches do start when device is powered.

**Reply by gert4git** [reply]
@Jason2866, no I am using the little clock example with the TFT. Compile & upload -> clock is on TFT. power-cycle TFT is black.
